rust/Cargo.toml

```diff
@@ -34,6 +34,7 @@ categories = ["database"]
 
 [workspace.dependencies]
 adbc_core = { path = "./core" }
+arrow = { version = "53.1.0", default-features = false }
```

Comment:
Instead of adding `arrow` we can add the `arrow` subcrates to reduce the size of our dependency tree.
It looks like we need to add:
- `arrow-cast` for https://docs.rs/arrow-cast/latest/arrow_cast/pretty/fn.print_batches.html (`dev-dependency`)
- `arrow-select` for https://docs.rs/arrow-select/latest/arrow_select/concat/fn.concat_batches.html

Reply:
thanks. makes sense, didn't realize most of these functions could be imported from multiple crates. Added both as a dev-dependency.